src/database/arangodb/aql-generator.ts

```diff
@@ -1399,153 +1418,842 @@ register(FollowEdgeQueryNode, (node, context) => {
+    const innerMapFrag = (itemFrag: AQLFragment) => {
+        const innerContext = context
+            .introduceVariableAlias(node.itemVariable, itemFrag)
+            .introduceVariableAlias(node.rootEntityVariable, rootVar);
+        const valueFrag = node.innerNode ? processNode(node.innerNode, innerContext) : itemFrag;
+
+        return aql.lines(
+            aql`{`,
+            aql.indent(
+                aql.lines(
+                    aql`value: ${valueFrag},`,
+                    aql`sortValues: [`,
+                    aql.indent(
+                        aql.join(
+                            node.orderBy.clauses.map((c) => processNode(c.valueNode, innerContext)),
+                            aql`,\n`,
+                        ),
+                    ),
+                    aql`]`,
+                ),
+            ),
+            aql`}`,
+        );
+    };
+
+    const filterNode = node.filterNode;
+    const innerFilterFrag = filterNode
+        ? (itemFrag: AQLFragment) => {
+              // don't provide rootEntityVariable
+              // (if we want to filter on root, it should happen outside already)
+              const innerContext = context.introduceVariableAlias(node.itemVariable, itemFrag);
+              return processNode(filterNode, innerContext);
+          }
+        : undefined;
+
+    const outerMapFrag = getFieldTraversalFragment({
+        segments: node.fieldSegments,
+        sourceFrag: rootVar,
+        mapFrag: innerMapFrag,
+        filterFrag: innerFilterFrag,
+    });
+
+    // holds the { value, sortValues }
+    const itemVar = aql.variable(`item`);
+
+    const clauseFrags = node.orderBy.clauses.map(
+        (clause, index) =>
+            aql`${itemVar}.sortValues[${aql.integer(index)}]${dirAQL(clause.direction)}`,
+    );
+
+    // The outerMapFrag will produce a list, so a simple RETURN mapFrag would result in nested lists
+    // -> we iterate over the items again to flatten the lists (the FOR ${itemVar} ...)
+    return aqlExt.parenthesizeList(
+        aql`${forStatementsFrag}`,
+        aql`FOR ${itemVar} IN ${outerMapFrag}`,
+        aql`SORT ${aql.join(clauseFrags, aql`, `)}`,
+        generateLimitClause(node) ?? aql``,
+        aql`RETURN ${itemVar}.value`,
+    );
+}
+
+/**
+ * Produces:
+ *
+ *     source.listFieldSegment1[*].listFieldSegment2[**][*
+ *         FILTER filterExpr(CURRENT.nonListFieldSegment)
+ *         LIMIT skip, maxCount
+ *         RETURN innerNode(CURRENT.nonListFieldSegment)
+ *     ]
+ */
+function processTraversalWithOnlyFieldSegmentsUsingArrayExpansion(
+    node: TraversalQueryNode,
+    context: QueryContext,
+): AQLFragment {
+    if (node.relationSegments.length) {
+        throw new Error(`Expected no relation segments`);
+    }
     if (node.entitiesIdentifierKind !== EntitiesIdentifierKind.ENTITY) {
         throw new Error(`Only ENTITY identifiers supported without relationSegments`);
     }
```

Comment:
I need some explanation what exactly the `entitiesIdentifierKind` is 😓

Reply:
it specifies whether `TraversalQueryNode.sourceEntityNode` holds the source entity object (`ENTITY`) or an ID (`ID`).